File: debug/Hello.cs
using System;

public class Hello {
    public static void PrintArray(string prefix, string[] words) {
        Console.WriteLine(prefix + ": " + String.Join(", ", words));
    }

    public static void Main(string[] args) {
        Console.WriteLine("Hello, World!");
        PrintArray("Before", args);
        Array.Sort(args);
        PrintArray("After", args);
    }
}

File: github.com/mindreframer/go-thestdlib/debug/hello.c
#include "stdio.h"
#include "stdlib.h"
#include "string.h"

int c(const void *l, const void *r) {
    return strcmp(*(char * const *)l, *(char * const *)r);
}

void print(int c, char **ss) {
    int i;
    for (i = 0; i < c; i++) {
        printf("%d: %s\n", i, ss[i]);
    }
}

int main(int argc, char **argv) {
    printf("Hello, World!\n");
    printf("ARGV before:\n");
    print(argc, argv);
    qsort(argv, argc, sizeof(char *), c);
    printf("ARGV after:\n");
    print(argc, argv);
    return 0;
}
